stl/inc/flat_map

```diff
@@ -0,0 +1,873 @@
+    }
+
+    void _Insert_exact(const_iterator _Position, value_type&& _Val) {
+        _Clear_flat_map_scope_guard _Guard{this};
+        _Data.keys.insert(_Position._Key_it, _STD move(_Val.first));
+        _Data.values.insert(_Position._Mapped_it, _STD move(_Val.second));
+        _Guard._Clearable = nullptr;
+    }
+
+    template <bool _Override_if_exists, class _K, class... _Mapped_args_t>
+    iterator _Emplace_hint(const_iterator _Position, _K&& _Key, _Mapped_args_t&&... _Args)
+        requires _STD
+    is_constructible_v<mapped_type, _Mapped_args_t...>
+        && (same_as<_STD remove_cvref_t<_K&&>, key_type>
+            || (constructible_from<key_type, _K&&> && _Compare_is_transparent<key_compare>) ) {
+        const const_iterator _Begin = cbegin();
+        const const_iterator _End   = cend();
+
+        // Adjust _Position
+        if (_Position == _End || !_Key_compare(*(_Position._Key_it), _Key)) {
+            // _Val should be inserted at or before _Position
+            if (_Position == _Begin || !_Key_compare(_Key, *((_Position - 1)._Key_it))) {
+                // _Val should be inserted at or after _Position
+                // The hint _Position is accurate
+            } else {
+                auto _Key_Position = _STD upper_bound(_Data.keys.cbegin(), _Position._Key_it, _Key, _Key_compare);
+                auto _Distance     = _STD distance(_Data.keys.cbegin(), _Key_Position);
+                _Position          = _Begin + _Distance;
+            }
+        } else {
+            auto _Key_Position = _STD lower_bound(_Position._Key_it + 1, _Data.keys.cend(), _Key, _Key_compare);
+            auto _Distance     = _STD distance(_Data.keys.cbegin(), _Key_Position);
+            _Position          = _Begin + _Distance;
+        }
+
+        bool _Should_insert = false;
+        if (_Position == _End) {
+            _Should_insert = true;
+        } else if (_Key_equal(*(_Position._Key_it), _Key)) {
+            _Should_insert = false;
+        } else {
+            _Should_insert = true;
+        }
+
+        auto _Distance = _STD distance(_Begin, _Position);
+        if (_Should_insert) {
+            _Insert_exact(_Position, value_type(_STD forward<_K>(_Key), _STD forward<_Mapped_args_t>(_Args)...));
+        } else if constexpr (_Override_if_exists) {
+            auto _It          = begin() + _Distance;
+            *(_It._Mapped_it) = mapped_type(_STD forward<_Mapped_args_t>(_Args)...);
+        }
+        // Consider _Position invalidated by the insert operation
+        return begin() + _Distance;
+    }
+
+    template <class _InputIterator, bool _NeedSorting, bool _NeedDeduping>
+        requires _Is_iterator_v<_InputIterator>
+    void _Insert_range(_InputIterator _First, _InputIterator _Last) {
+        _Clear_flat_map_scope_guard _Guard{this};
+
+        // Insert the new elements at the end
+        size_t _OldSize = size();
+        size_t _NewSize = _STD distance(_First, _Last);

```

Comment:
Though no actual difference, better be `size_type` as they `looks` safer.

Reply:
Fixed.